packages/postgres-database/src/simcore_postgres_database/utils_groups_extra_properties.py

```diff
@@ -0,0 +1,150 @@
+import datetime
+from dataclasses import dataclass, fields
+from typing import Any
+
+import sqlalchemy
+from aiopg.sa.connection import SAConnection
+from aiopg.sa.result import RowProxy
+from simcore_postgres_database.models.groups_extra_properties import (
+    groups_extra_properties,
+)
+
+from .models.groups import GroupType, groups, user_to_groups
+from .utils_models import FromRowMixin
+
+
+class GroupExtraPropertiesError(Exception):
+    ...
+
+
+class GroupExtraPropertiesNotFound(GroupExtraPropertiesError):
+    ...
+
+
+@dataclass(frozen=True, slots=True, kw_only=True)
+class GroupExtraProperties(FromRowMixin):
+    group_id: int
+    product_name: str
+    internet_access: bool
+    override_services_specifications: bool
+    created: datetime.datetime
+    modified: datetime.datetime
+
+
+async def _list_table_entries_ordered_by_group_type(
+    connection: SAConnection, user_id: int, product_name: str
+) -> list[RowProxy]:
+    subquery = (
+        sqlalchemy.select(
+            *[
+                groups_extra_properties,
+                groups.c.type,
+                sqlalchemy.case(
+                    *[
+                        # NOTE: the ordering is important for the aggregation afterwards
+                        (groups.c.type == "EVERYONE", sqlalchemy.literal(3)),
+                        (groups.c.type == "STANDARD", sqlalchemy.literal(2)),
+                        (groups.c.type == "PRIMARY", sqlalchemy.literal(1)),
+                    ],
+                    else_=sqlalchemy.literal(4),
```

Comment:
Q: this is here so the query is robust, or there are situations where we do not have the `type` column prefilled?

Reply:
no there are no such things. It is more for robustness in case it happens later.